customer-errors.go

```diff
@@ -17,3 +17,19 @@ func NewErrCustomerNotFound(id int64) ErrCustomerNotFound {
 		id,
 	}
 }
+
+// ErrCustomerSearchNotFound is thrown when no customer is found for the given search query
+type ErrCustomerSearchNotFound struct {
+	query string
+}
+
+func (err ErrCustomerSearchNotFound) Error() string {
+	return fmt.Sprintf("customer from query `%v` was not found", err.query)
+}
+
+// ErrCustomerSearchNotFound builds the error
+func NewErrCustomerSearchNotFound(query string) ErrCustomerSearchNotFound {
+	return ErrCustomerSearchNotFound{
+		query,
+	}
+}
```

Comment:
The query does not need to return a customer if the query doesn't find one, this seems valid/ not an error :thinking: